R/SS_write.R

```diff
@@ -0,0 +1,110 @@
+#' Write all Stock Synthesis input files for a model
+#'
+#' Writes all the input files for a Stock Synthesis model using the list
+#' created by [SS_read()]
+#' (presumably after modification of one or more elements)
+#' using the `SS_write*()` functions for the four or
+#' five model input files.
+#'
+#' @param inputs list created by [SS_read()]
+#' @template dir
+#' @param files vector of file types to write. The default is the four
+#' standard input files plus wtatage.ss if used. The ss.par file is not
+#' yet supported by this function or [SS_read()].
+#' @template overwrite
+#' @template verbose
+#'
+#' @author Ian G. Taylor
+#'
+#' @export
+#' @seealso
+#' * [SS_read()] creates the list that is used by this function.
+#' * [SS_writestarter()], [SS_writedat()], [SS_writectl()],
+#' [SS_writeforecast()], and [SS_writewtatage()] are used to write the
+#' input files.
+
+#' @examples
+#' \dontrun{
+#' # read inputs to modify the data file
+#' inputs <- SS_read(
+#'   dir = system.file("extdata", "simple_3.30.13", package = "r4ss")
+#' )
+#'
+#' # modify the starter file (use the par file)
+#' inputs$start$init_values_src <- 1
+#'
+#' # modify the data file (remove age comps from years prior to 1990)
+#' inputs$dat$agecomp <- inputs$dat$agecomp %>% dplyr::filter(Yr >= 1990)
+#'
+#' # modify the control file (turn off early recdevs and change range of yrs)
+#' inputs$ctl$recdev_early_phase <- -abs(inputs$ctl$recdev_early_phase)
+#' inputs$ctl$MainRdevYrFirst <- 1980
+#'
+#' # write the files to a new folder within the source directory
+#' SS_write(inputs = inputs,
+#'          dir = file.path(inputs$dir, "modified_inputs"))
+#' }
+
+SS_write <- function(inputs,
+                     dir = NULL,
+                     files = c("dat", "ctl", "start", "fore", "wtatage"),
+                     overwrite = FALSE,
+                     verbose = FALSE) {
+  # create directory if not already there
+  if (!is.null(dir)) {
+    if (!dir.exists(dir)) {
+      if (verbose) {
+        message("Creating new directory: ", dir)
+      }
+      dir.create(dir)
+    }
+  } else {
+    dir <- ""
```

Comment:
Why make this a hidden input? Couldn't you just make the default ""? It seems like it would be more transparent to the user that way.